### PR TITLE
Implement new core foundation

### LINKED ISSUE #46: Implementing new core foundation
Currently I'm working on a new core foundation for the library based on nodes representation for the rich text.
Here's some benefits of this change:
- Improve scalability of the library
- Implement new features more easily
- Bug fixes
- Improve HTML output (Shorter)
- Performance improvements
...

This update will bring breaking changes so we are going to move from 0.x to 1.x
The work is in its last stages, it will be ready soon

### PR DESCRIPTION
**Pull request by MohamedRejeb**
Currently I'm working on a new core foundation for the library based on nodes representation for the rich text.
Here's some benefits of this change:

- Improve scalability of the library
- Implement new features more easily
- Bug fixes
- Improve HTML output (Shorter)
- Performance improvements
...
This update will bring breaking changes so we are going to move from 0.x to 1.x
The work is in its last stages, it will be ready soon

Closes #46

### CODE REVIEW

File: richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichSpanStyle.kt [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+        return index to richSpanStyleList
+    }
+
+    fun removeTextRange(textRange: TextRange): RichSpanStyle? {
+        if (textRange.min <= this.textRange.min && textRange.max >= fullTextRange.max) return null
+
+        // Remove text from start
+        if (textRange.min in this.textRange || (textRange.max - 1) in this.textRange) {
+            println("Remove text from start")
```
Comment (DAKSHSEMWAL) [inline richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichSpanStyle.kt:109]: I guess these will be removed

File: richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichSpanStyle.kt [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+            println("textRange: $textRange")
+            println("this.textRange: ${this.textRange}")
+            println("text: ${this.text}")
```
Comment (DAKSHSEMWAL) [inline richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichSpanStyle.kt:112]: I guess these will be removed

Comment (MohamedRejeb) [reply richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichSpanStyle.kt:112]: Yes, it's not ready yet for merge. I'm still adding few things here and there.

**Review by DAKSHSEMWAL — APPROVED** [review]

**Comment by wakaztahir** [comment]
Would really appreciate if big pull requests would merge quickly so I can work on the library
I can help contribute a markdown parser library which would covert markdown to your AST , I hope this AST is a separated library , This way conversion from html to this AST can be done in a separate library 

I am hoping that library structure would look like this Core -> Contains AST Nodes that can be fed to the Rendering Library , Only dependency can be compose runtime , HTML -> Converts html code to AST and provides helper Composables to render HTML , Markdown -> Converts Markdown to AST and provides helper Composables to render Markdown

Initially I created this library https://github.com/Qawaz/markdown-compose , for one my apps , but since compose did not support Annotated String Save , The library didn't work , My plan is to take good parts of it and put in this library without causing any breaking changes , This library has code highlighting support through an extension , Also has mathjax support (desktop and android only) through extension

**Comment by MohamedRejeb** [comment]
I'm trying to merge this PR ASAP. For the markdown feature the most important part is a markdown parser, something similar to my html parser that I'm using, so it could be something totally separate from the library.

https://github.com/MohamedRejeb/Ksoup

**Pull request merged**